modules/Bio/EnsEMBL/Variation/StructuralVariationSample.pm

```diff
@@ -118,8 +121,14 @@ sub new {
   my $class = ref($caller) || $caller;
   my $self = $class->SUPER::new(@_);
 
-  my ($dbID,$adaptor,$structural_variation_id,$sample,$strain_id,$strain,$study_id, $study) =
-    rearrange([qw(dbID ADAPTOR _STRUCTURAL_VARIATION_ID SAMPLE _STRAIN_ID STRAIN _STUDY_ID STUDY)],@_); 
+  my ($dbID,$adaptor,$structural_variation_id,$sample,$strain_id,$strain,$study_id, $study, $zygosity) =
+    rearrange([qw(dbID ADAPTOR _STRUCTURAL_VARIATION_ID SAMPLE _STRAIN_ID STRAIN _STUDY_ID STUDY ZYGOSITY)],@_); 
+
+  if (defined($zygosity)) {
+    unless (grep $_ eq $zygosity, ('homozygous', 'heterozygous')) {
+      throw('Zygosity must be one of "homozygous" or "heterozygous"');
```

Comment:
Is this value set for zygosity sufficient? It does not account for male on X or Y, for example, or for other polyzygotic species which may want to use this API.

Elsewhere in the API we use an integer value for ploidy that represents the total number of chromosomes. This is cleaner for the backend and more extensible; the equivalent for zygosity could be easily nice-ified for web or whatever e.g. converting 1->het, 2->hom.

This is up for discussion though, thoughts @sarahhunt ?

Reply:
@willmclaren, yes you are right, we could use numbers instead of a set of terms.
At the moment the use case for us is quite limited, but I don't know if it will be used for other non diploide species